boards: shields: Clean up Kconfig.defconfig files

erwango opened the pull request:
Shield's Kconfig.defconfig files should only be used to configure components options, or drivers specific (eg init priority, ..), depending on driver activation which remains application responsibility. High level components (such as DISPLAY) should not be enabled by default on boards activation (which is similar to policy used for boards).

Additionally, clean up busses activation (SPI, I2C), which is not required as bus activation is already done automatically on bus devices activation.

erwango committed:
boards: shields: Clean up Kconfig.defconfig files

Shield's Kconfig.defconfig files should only be used to configure
components options, or drivers specific (eg init priority, ..), depending
on driver activation which remains application responsibility.
High level components (such as DISPLAY) should not be enabled by default on
boards activation (which is similar to policy used for boards).

Additionally, clean up busses activation (SPI, I2C), which is not required
as bus activation is already done automatically on bus devices activation.

Signed-off-by: Erwan Gouriou <erwan.gouriou@linaro.org>

gmarull commented on boards/shields/npm1100_ek/Kconfig.defconfig:
```diff
@@ -1,9 +0,0 @@
-# Copyright (c) 2023 Nordic Semiconductor ASA
-# SPDX-License-Identifier: Apache-2.0
-
-if SHIELD_NPM1100_EK
-
-config REGULATOR
-	default y
-
```
Indeed, you're totally right.

gmarull commented on boards/shields/npm1100_ek/Kconfig.defconfig:
```diff
@@ -1,9 +0,0 @@
-
-if SHIELD_NPM1100_EK
-
-config REGULATOR
```
if it's not too much to ask, could you add CONFIG_REGULATOR=y to https://github.com/zephyrproject-rtos/zephyr/blob/main/samples/shields/npm6001_ek/prj.conf

erwango replied:
Sure, thanks for the heads up.

erwango committed:
sample: shields: Update prj.conf

Before Kconfig clean up to be performed in boards/shields, update samples
to enable the components they require.

Signed-off-by: Erwan Gouriou <erwan.gouriou@linaro.org>

gmarull commented on samples/shields/npm1300_ek/prj.conf:
```diff
@@ -3,3 +3,4 @@
 
 CONFIG_SHELL=y
 CONFIG_LOG=y
+CONFIG_REGULATOR=y
```
missing gpio and sensor here

erwango replied:
`CONFIG_SENSOR=y` added, but I omit GPIO, which is part of default boards configuration (so theoretically should always be there).
But I can add it, if you think there's a risk it is finally missing.

gmarull commented on samples/shields/npm6001_ek/prj.conf:
```diff
@@ -5,3 +5,4 @@ CONFIG_SHELL=y
 CONFIG_SHELL_GETOPT=y
 CONFIG_GETOPT=y
 CONFIG_GETOPT_LONG=y
+CONFIG_REGULATOR=y
```
`CONFIG_GPIO=y`, `CONFIG_WATCHDOG=y`

erwango commented on boards/shields/max3421e/Kconfig.defconfig:
```diff
@@ -1,9 +0,0 @@
-
-if SHIELD_SPARKFUN_MAX3421E
-
-config UHC_DRIVER
```
@jfischer-no I'd like you to verify on this specific change.
To me, there is no reason this driver doesn't follow the same rule as others

jfischer-no replied:
Change is correct, thanks.

erwango committed:
boards: sam_v71_xult: Enable GPIO Kconfig

CONFIG_GPIO is expected to be enabled by default.

Signed-off-by: Erwan Gouriou <erwan.gouriou@linaro.org>
